### PR TITLE
Texture: Add Texture.DEFAULT_COLOR_SPACE

### PR DESCRIPTION
**Pull request by linbingquan**
Related issue: #XXXX

**Description**

Add `Texture.DEFAULT_COLORSPACE` variable.
In some cases, reduce the amount of duplicate code.

### CODE REVIEW

**Comment by donmccurdy** [comment]
Interesting idea! Let's use the `DEFAULT_COLOR_SPACE` (three words) for consistency with `.colorSpace`. Note that we'd also need to update the loaders...

https://github.com/mrdoob/three.js/blob/ed211d89c647234e94394f92abffab60cfe3db8f/examples/jsm/loaders/GLTFLoader.js#L3374-L3378

... otherwise they'll assume the default is NoColorSpace, and fail to set up non-color textures (like `.normalMap`) correctly. It would also be important to clarify... is this meant to be the default for all textures when not otherwise specified? Or the default for all _color_ textures? If you set `Texture.DEFAULT_COLOR_SPACE = THREE.DisplayP3ColorSpace`, would you expect loaders to ...

- (A) assign color textures Display P3 (from the `DEFAULT_COLOR_SPACE`), and non-color textures NoColorSpace
- (B) assign color textures sRGB (because it's standard for most model formats), and non-color textures NoColorSpace
- (C) assign all textures `DEFAULT_COLOR_SPACE`
- (D) ... something else?

**Comment by linbingquan** [comment]
Thank you for your response.

> Interesting idea! Let's use the `DEFAULT_COLOR_SPACE` (three words) for consistency with `.colorSpace`.

Done.

> Note that we'd also need to update the loaders...

TBH, I don't know which loaders should be updated...

I've seen examples with quite a bit of duplicate code like: `texture.colorSpace = THREE.SRGBColorSpace;`

I found three loaders(EXRLoader, KTX2Loader and USDZLoader) that reference `NoColorSpace`.

**Comment by donmccurdy** [comment]
> I've seen examples with quite a bit of duplicate code like: texture.colorSpace = THREE.SRGBColorSpace;

Personally I would recommend that we keep the examples this way, perhaps unless they're intentionally using a wide gamut like P3 rather than sRGB, which is only experimental right now. I worry that if we 'simplify' the examples by doing `Texture.DEFAULT_COLOR_SPACE = SRGBColorSpace` we're  going to lead users down a more confusing path, where other textures (like normal maps, aoMaps, metal/rough maps ...) don't look right, presenting weird and confusing issues. Whereas if the color texture has the wrong color space, it's considerably easier to debug.

We'll need to check all the loaders in https://github.com/mrdoob/three.js/issues/23283, if they support textures, but it doesn't need to be in this PR. The goal would be to make sure the loaders always specify the `.colorSpace` property, using DEFAULT_COLOR_SPACE (or whatever the format dictates) for color textures, and NoColorspace for non-color textures.

/cc @gkjohnson @WestLangley

File: src/textures/Texture.js [hunk 1]
```diff
@@ -316,5 +316,6 @@ class Texture extends EventDispatcher {
 Texture.DEFAULT_IMAGE = null;
 Texture.DEFAULT_MAPPING = UVMapping;
 Texture.DEFAULT_ANISOTROPY = 1;
+Texture.DEFAULT_COLOR_SPACE = NoColorSpace;
```
Comment (WestLangley) [inline src/textures/Texture.js:319]: Setting the "default color space" to "not a color space" seems extremely odd to me.

**Comment by gkjohnson** [comment]
Generally I'm very against these kinds of editable global flags since they make function behavior less predictable and less reliable. They just pass on the verbosity to dependent projects. Ie if I have a project and I need loaded or processed data to be set to `NoColorSpace` in some other dependency then I need to do the following every time:

```js
const originalDefaultColorSpace = Texture.DEFAULT_COLOR_SPACE;
Texture.DEFAULT_COLOR_SPACE = NoColorSpace;

// load or generate the data

Texture.DEFAULT_COLOR_SPACE = originalDefaultColorSpace;
```

It also sets projects to break more easily. For example if someone is using three-gpu-pathtracer or the mesh bvh shader textures and they've set the default texture color space to sRGB then suddenly all my path tracer data textures break because they're implicitly set to the wrong color space.

I think some examples of what this is actually helping should be provided so we can determine if this is actually addressing an issue and discuss other options. Without knowing more it should be very simple for end users to create utility functions that set the appropriate color space.

**Comment by WestLangley** [comment]
I understand the motivation behind this PR, but I think the implications of this change would be too confusing.

**Comment by linbingquan** [comment]
To be honest, I didn't think much of it, and after reading some feedback, I found that adding `Texture.DEFAULT_COLOR_SPACE` was confusing.  Because it involves other things like loaders and other libs dependencie on existing `Texture.colorSpace` paramater. I feel this PR can be close.

**Comment by donmccurdy** [comment]
Ok, I believe you are right – thanks @gkjohnson and @WestLangley!

**Pull request closed without merging**